app/model/base.py

```diff
@@ -0,0 +1,48 @@
+"""Define the abstract base class for a data model."""
+from abc import ABC, abstractmethod
+from typing import Dict, Any
+
+
+class RocketModel(ABC):
+    """Define the properties and methods needed for a data model."""
+
+    @abstractmethod
+    def get_attachment(self) -> Dict[str, Any]:
+        """Return slack-formatted attachment (dictionary) for data model."""
+        pass
+
+    @staticmethod
+    @abstractmethod
+    def to_dict(model: 'Model') -> Dict[str, Any]:

```

Comment:
```suggestion
    def to_dict(model: 'RocketModel') -> Dict[str, Any]:
```

For every instance of `Model`.